File: GENERICIDAD/Ejercicio 1/codigo/codigo.py
#1. Crea una clase genérica Caja<T> para guardar algún tipo de objeto
#   a) Agrega métodos guardar() y obtener()
#   b) Crea dos instancias de la caja y almacena 2 datos de diferente tipo
#   c) Muestra el contenido de las cajas
class Caja:
    def __init__(self):
        self.contenido = None
    
    def guardar(self, item):
        self.contenido = item
    
    def obtener(self):
        return self.contenido

if __name__ == "__main__":
    caja_texto = Caja()
    caja_texto.guardar("Contenido de texto")
    caja_numero = Caja()
    caja_numero.guardar(100)
    print(f"Caja de texto: {caja_texto.obtener()}")
    print(f"Caja de número: {caja_numero.obtener()}")

File: GENERICIDAD/Ejercicio 1/codigo/codigo.java
//1. Crea una clase genérica Caja<T> para guardar algún tipo de objeto
//  a) Agrega métodos guardar() y obtener()
//  b) Crea dos instancias de la caja y almacena 2 datos de diferente tipo
//  c) Muestra el contenido de las cajas
public class Caja<T> {
    private T contenido;
    public void guardar(T item) {
        this.contenido = item;
    }

    public T obtener() {
        return contenido;
    }

    public static void main(String[] args) {
        Caja<String> cajaTexto = new Caja<>();
        cajaTexto.guardar("Contenido de texto");
        Caja<Integer> cajaNumero = new Caja<>();
        cajaNumero.guardar(100);
        System.out.println("Caja de texto: " + cajaTexto.obtener());
        System.out.println("Caja de número: " + cajaNumero.obtener());
    }
}